csharp-advanced-topics/WeakEventsInCSharp/Tests/Tests.cs

```diff
@@ -0,0 +1,48 @@
+using WeakEventsInCSharp;
+
+namespace Tests;
+
+public class Tests
+{
+    private bool _eventHandled;
+    private void EventHandlerMethod(object sender, EventArgs e)
+    {
+        _eventHandled = true;
+    }
+
+    [Fact]
+    public void GivenStrongEventWhenGCCollectedThenEventRaised()
+    {
+        var publisher = new Publisher();
+        var subscriber = new Subscriber();
+
+        publisher.Event += EventHandlerMethod;
+        publisher.RaiseEvent();
+
+        Assert.True(_eventHandled);
+
+        subscriber = null;
+        GC.Collect();
+        publisher.RaiseEvent();
+
+        Assert.True(_eventHandled);
+    }
+
+    [Fact]
+    public void GivenWeakEventWhenGCCollectedThenEventNotRaised()
+    {
+        var weakEventPublisher = new WeakReferencePublisher();
+        var weakEventSubscriber = new WeakReferenceSubscriber();
+        weakEventSubscriber.Subscribe(weakEventPublisher);
+        weakEventPublisher.RaiseEvent();
+
+        Assert.True(weakEventPublisher.Event.HandlerInvoked);
+
+        weakEventSubscriber = null;
```

Comment:
When I comment out this line, and the `subscriber = null` line in the other test, both tests pass anyway.

It seems like calling `GC.Collect()` on the weak reference is enough for it to dispose of it, even if it hasn't been set to `null`. Do you think that's expected behavior?

Reply:
The explicit null set is more to highlight the intent that the subscriber is being removed. Without a strong reference present to the weakEventSubscriber, GC will be able to collect it without the explicit null set.

I think this is expected.